### PR TITLE
OCPQE-14026:  134_IPI on AWS & KMS & Etcd Encryption.install.yaml

### PR DESCRIPTION
**Pull request by yunjiang29**
Migrate profile 134_IPI on AWS & KMS & Etcd Encryption.install.yaml to Prow CI

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
@jianlinliu PTAL.

**Comment by yunjiang29** [comment]
@Phaow please also take a look to check if it satisfies your requirements on KMS generation and usage.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/byo-kms/etcd-encryption/provision/cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-provision-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+chain:
+  as: cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-provision
+  steps:
+  - ref: aws-provision-vpc-shared
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/byo-kms/etcd-encryption/provision/cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-provision-chain.yaml:4]: In this case, do we have to combine customer vpc?

Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/byo-kms/etcd-encryption/provision/cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-provision-chain.yaml:4]: ahh, you're right, will remove it.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/byo-kms/etcd-encryption/deprovision/cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-deprovision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+chain:
+  as: cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-deprovision
+  steps:
+  - chain: cucushift-installer-rehearse-aws-ipi-deprovision
+  - ref: aws-deprovision-stacks
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/byo-kms/etcd-encryption/deprovision/cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-deprovision-chain.yaml:5]: We do not need this step any more, right?

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.12.yaml [hunk 3]
```diff
@@ -39,7 +39,7 @@ tests:
     cluster_profile: aws-qe
     env:
       BASE_DOMAIN: qe.devcluster.openshift.com
-    workflow: cucushift-installer-rehearse-aws-ipi-ingress-lbtype-classic
+    workflow: cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.12.yaml:42]: We will have a followup PR to add the coverage into ci job yaml file, right?

Comment (yunjiang29) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.12.yaml:42]: yes, for 4.11-4.12

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/byo-kms/etcd-encryption/deprovision/cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-deprovision-chain.yaml (hunk 2 above)
Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/byo-kms/etcd-encryption/deprovision/cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-deprovision-chain.yaml:5]: fixed.

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by Phaow** [comment]
> @Phaow please also take a look to check if it satisfies your requirements on KMS generation and usage.

Thanks for the migration! From storage scope looks we need to get the kms keyID and add a policy to the ebs-csi-driver-controller user as the key user. I think I could add the extra steps which storage needed later after I do some test related https://issues.redhat.com/browse/STOR-870

**Comment by yunjiang29** [comment]
> Thanks for the migration! From storage scope looks we need to get the kms keyID and add a policy to the ebs-csi-driver-controller user as the key user. I think I could add the extra steps which storage needed later after I do some test related https://issues.redhat.com/browse/STOR-870

@Phaow current steps don't support:
1. generate two different KMS keys
2. custom key policy while generating KMS key.

But `ipi-conf-aws-kms-key` does support user providing two different existing KMS keys, I think you can use the _permanent_ KMS key with your own policy.

Are the current steps good for you or do any other capabilities need to be added?

**Comment by Phaow** [comment]
> Are the current steps good for you or do any other capabilities need to be added?

@yunjiang29 Sorry for late reply. Thanks for helping migrating the profile again! Just double checked with our DEV if [OCPBUGS-5410](https://issues.redhat.com/browse/OCPBUGS-5410) fixed by modifying the CR credentialsrequest/aws-ebs-csi-driver-operator current profile should satisfy all storage requirements. If it just a doc fix, I need to change the kmskey's policy or add more permission to our operator user in aws.

**Comment by yunjiang29** [comment]
@Phaow np, just let me know if you need to update `aws-provision-kms-key` or `ipi-conf-aws-kms-key`

**Comment by Phaow** [comment]
> @Phaow np, just let me know if you need to update `aws-provision-kms-key` or `ipi-conf-aws-kms-key`

Sure, thanks! They are not needed to update currently.

**Pull request merged**